vcx/dummy-cloud-agent/src/indy/wallet.rs

```diff
@@ -16,4 +21,55 @@ pub fn open_wallet(config: &str, credentials: &str) -> Box<Future<Item=i32, Erro
 pub fn close_wallet(wallet_handle: i32) -> Box<Future<Item=(), Error=IndyError>> {
     wallet::close_wallet(wallet_handle)
         .into_box()
+}
+
+
+
+pub fn get_postgres_storage_plugin() -> String {
+    let os = os_type::current_platform();
+    let osfile = match os.os_type {
+        os_type::OSType::OSX => "/usr/local/lib/libindystrgpostgres.dylib",
+        _ => "/usr/lib/libindystrgpostgres.so"
+    };
+    return osfile.to_owned();
+}
+
+#[cfg(all(unix, test))]
+fn _load_lib(library: &str) -> libloading::Result<libloading::Library> {
+    libloading::os::unix::Library::open(Some(library), ::libc::RTLD_NOW | ::libc::RTLD_NODELETE)
+        .map(libloading::Library::from)
+}
+
+#[cfg(any(not(unix), not(test)))]
+fn _load_lib(library: &str) -> libloading::Result<libloading::Library> {
+    libloading::Library::new(library)
+}
+
+pub fn load_storage_library(library: &str, initializer: &str, storage_config: &str, storage_credentials: &str){
+    debug!("Loading storage plugin '{:}' as dynamic library.", library);
+    let lib_res = _load_lib(library);
+    match lib_res {
+        Ok(lib) => {
+            unsafe {
+                debug!("Storage library '{:}' loaded. Resolving its init function '{:}'.", library, initializer);
+                let init_func: libloading::Symbol<unsafe extern fn() -> ErrorCode> = lib.get(initializer.as_bytes()).unwrap();
```

Comment:
avoid unwrap and panic usages. 
It is better to return Result and handle error above like (Storage loading failed with: {:?}).